relay-server/src/utils/statsd.rs

```diff
@@ -58,6 +69,12 @@ where
         changes = changes,
         source_out = new_source.as_str(),
     );
+    relay_statsd::metric!(
+        set(RelaySets::TransactionNameChanges) = old_name,
```

Comment:
No, it will buffer a bit more data, but the amount of tags is the same as for the counter metric. In the end, it should only transmit one unique count per flush: https://github.com/statsd/statsd/blob/master/docs/metric_types.md#sets.